### ISSUE TITLE
Unable to create request URL, because OpenWhisk API host is missing

### ISSUE THREAD

**Issue by tleyden** [apache/incubator-openwhisk-client-go#43]
When I build against version of https://github.com/apache/incubator-openwhisk-client-go/commit/ad5aa3402b033edd8b30aec7f33434a253aa35ed and run on IBM Cloud, [this code](https://github.com/tleyden/keynuker/blob/110c0a672b75a4ccf5331bc08f2fc3fe354ac1c1/openwhisk_status.go#L111) is getting an error:

```
Unable to create request URL, because OpenWhisk API host is missing"
```

even though `__OW_API_HOST` and `__OW_API_KEY` are set to correct values in the environment (which is expected since it's running on IBM Cloud)

This same code was working fine against an older commit (https://github.com/apache/incubator-openwhisk-client-go/commit/b3d194d8b937a801ee11efcf7f2544d9cea1f58e), so I think something changed in this library.

What updates do I need to make in my code to make this work?

Currently my code is [directly loading the environment variables](https://github.com/tleyden/keynuker/blob/110c0a672b75a4ccf5331bc08f2fc3fe354ac1c1/openwhisk_status.go#L246-L268) and passing that to the `whisk.NewClient()` call.  Is that still the right way to do things?  I believe I did that as a workaround to https://github.com/apache/incubator-openwhisk-client-go/issues/23

**Comment by tleyden** [comment]
As an experiment, I tried changing:

```
client, err := whisk.NewClient(http.DefaultClient, whiskConfig)
```

(where the `whiskConfig` struct is built up via [this code](https://github.com/tleyden/keynuker/blob/110c0a672b75a4ccf5331bc08f2fc3fe354ac1c1/openwhisk_status.go#L246-L268))

to what's in the README:

```
client, err := whisk.NewClient(http.DefaultClient, nil)
```

and I'm getting this error:

> OpenWhisk API host is missing (Please configure whisk.api.host.proto, whisk.api.host.name and whisk.api.host.port in whisk.properties under the OPENWHISK_HOME directory.)


How do those configuration instructions translate to running on IBM Cloud?  I don't believe there is any direct access to `whisk.properties` AFAIK in that environment.

**Comment by csantanapr** [comment]
Hum that's weird we refer to whisk.properties directly from the go code

**Comment by csantanapr** [comment]
You point out IBM cloud not working, does that mean that it works when you deploy your own OpenWhisk from open source ?

**Comment by tleyden** [comment]
I haven't tried that

**Comment by csantanapr** [comment]
When is nil what are the values use for authkey and apihost?
I think should always expect a whiskConfig object and never nil

**Comment by csantanapr** [comment]
I  see a getDefaultConfig when nil is that suppose to load the settings from __OW... ?
 https://github.com/apache/incubator-openwhisk-client-go/blob/master/whisk/client.go

**Comment by tleyden** [comment]
Digging into this a bit more, I think I found a problem with the way my code is constructing the `whiskConfig` struct: it's setting the `BaseURL` field, but not the `Host` field, and it looks like the [code](https://github.com/apache/incubator-openwhisk-client-go/blob/master/whisk/client.go#L140) expects the `Host` field to be set.

**Comment by csantanapr** [comment]
Maybe that could fue recent change and README was not updated to reflect changes

**Comment by tleyden** [comment]
Yeah I think the README should be updated to mention that the `whisk.Config.Host` field is required.

I just verified that [setting the whisk.Config.Host field](https://github.com/tleyden/keynuker/commit/bf1a5712649061a6b57b68990ce21a120b1fa1b0) fixes it on IBM Cloud

I think from a user perspective the when using this library on IBM Cloud, the ideal usage would be:

```
config := whisk.NewConfig()
config.InitFromEnvironment()
client, err := whisk.Newclient(http.DefaultClient, config)
```

and `InitFromEnvironment()` would effectively be doing the same thing as [WhiskConfigFromOwEnvVars()](https://github.com/tleyden/keynuker/blob/bf1a5712649061a6b57b68990ce21a120b1fa1b0/openwhisk_status.go#L247-L271)

**Comment by csantanapr** [comment]
@tleyden yeah I think that would be good addition.
But this environment variables are not specific to IBM, they will work on any OpenWhisk deployment

**Comment by houshengbo** [comment]
@tleyden Reading the information of credentials from whisk.properties is deprecated now, but still supported.

We currently supports to read the credentials from $HOME/.wskprops.
You can configure the following information:
* APIHOST,  AUTH and NAMESPACE, which are mandatory.
* APIGW_ACCESS_TOKEN,  APIVERSION,  KEY and CERT, which are optional.

Users do not need to care BaseURL any more. Instead, APIHOST is an easier configuration.

Reading credentials from environmental variables are not 100% guaranteed. The best way to to configure the $HOME/.wskprops.

**Comment by drcariel** [comment]
To add: I ran into this problem a few weeks ago, but only when building from the cli specific repos. However, the cli/go client code within incubator-openwhisk did not have this problem. Maybe you should investigate there @tleyden , the code _may_ be out of sync.

**Comment by tleyden** [comment]
@houshengbo Thanks for the list of mandatory and optional fields.  I think the library would be a bit easier to use if that was documented in the [exported Config struct](https://github.com/apache/incubator-openwhisk-client-go/blob/master/whisk/client.go#L70-L82).

> Reading credentials from environmental variables are not 100% guaranteed. The best way to to configure the $HOME/.wskprops.

 I'm invoking this code from within an OpenWhisk action, which is running on IBM Cloud (though according to @csantanapr, the same would be true on any OpenWhisk installation).  So reading from `$HOME/.wskprops` isn't an option.

**Comment by tleyden** [comment]
@drcariel the CLI is running in a different context, and it can just expect to be able to check local files like `$HOME/.wskprops`.  But in the context of a running action, the only thing available is environment variables.

Btw, I'm using an action to monitor walk the list of recent activations to look for failures, and then send alerts if there are any.
